AnimatedComponents/src/components/floating-button/floating-button-with-label/FloatingButtonWithLabel.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react'
+import { Image, Pressable, View } from 'react-native'
+import Animated, {
+  Easing,
+  Extrapolation,
+  SharedValue,
+  interpolate,
+  useAnimatedStyle,
+  useDerivedValue,
+  useSharedValue,
+  withDelay,
+  withSpring,
+  withTiming,
+} from 'react-native-reanimated'
+
+import { PlusIcon } from '@assets'
+import { FloatingButtonProps } from '@types'
+
+import styles from './floatingButtonWithLabel-styles'
+
+const FloatingButtonWithLabel = (props: FloatingButtonProps) => {
+  const {
+    animationTransitionDuration = 500,
+    buttonContainerStyle = styles.buttonContainer,
+    contentContainerStyle = styles.contentContainer,
+    icons,
+    iconContainerStyle = styles.iconContainer,
+    iconStyle = styles.icon,
+    textStyle = styles.text,
+    isLeft = false,
+  } = props
+
+  const isOpen = useSharedValue(false)
+  const opacity = useSharedValue(0)
+  const progress = useDerivedValue(() => (isOpen.value ? withTiming(1) : withTiming(0)))
+  const length = icons?.length
```

Comment:
naming ist iconLength or something realvant to it plz check
also why icon?.length as its mandatory prop icon.length should be fine. check.